cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,563 @@
+
+        // Assert that the play button is rendered
+        cy.get(`${wrapper}[data-testid='play-pause-button'][aria-label='Play']`).should("exist");
+
+        // Enable group layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+
+        // 243 + 12px + 12px = 267px
+        // See _MediaBody.pcss and _AudioPlayer.pcss for spacing
+        cy.get(`${wrapper}.mx_MAudioBody`).percySnapshotElement(detail + " on group layout", {
+            percyCSS,
+            widths: [267],
+        });
+
+        cy.log("Took a snapshot of " + detail + " on group layout");
+
+        // Enable bubble layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Bubble);
+
+        // 243px + 12px + 48px = 303px
+        // See _EventBubbleTile.pcss and _AudioPlayer.pcss for spacing
+        cy.get(`${wrapper}.mx_MAudioBody`).percySnapshotElement(detail + " on bubble layout", {
+            percyCSS,
+            widths: [303],
+        });
+
+        cy.log("Took a snapshot of " + detail + " on bubble layout");
+
+        // Reset the layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+    };
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, TEST_USER).then(() =>
+                cy.createRoom({}).then((_roomId) => {
+                    roomId = _roomId;
+                }),
+            );
+        });
+
+        cy.injectAxe();
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should render player on every layout", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Click the timestamp to highlight the event tile on which player is rendered
+            cy.get(".mx_EventTile .mx_MessageTimestamp").click();
+            cy.get(".mx_EventTile.mx_EventTile_selected .mx_MAudioBody").should("be.visible");
+
+            // Check audio player on IRC layout here, which currently should be same as on modern layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.IRC);
+            // 243 + 12px + 12px = 267px
+            // See _MediaBody.pcss and _AudioPlayer.pcss for spacing
+            cy.get(".mx_MAudioBody").percySnapshotElement("Audio player (light theme) on IRC layout", {
+                percyCSS,
+                widths: [267],
+            });
+            cy.log("Took a snapshot of Audio player (light theme) on IRC layout");
```

Comment:
I don't really understand why `takeSnapshots` doesn't do this. Could you help me understand?

Reply:
Sure! I did not include a snapshot test of the player on IRC layout in `takeSnapshots`, because the player's layout on IRC layout should be same as modern/group layout, and therefore the number of snapshots on IRC layout should be fine to be minimized, in order to keep the number of credits used per each test as low as possible.

Considering the fact that the tests on this file should consume a lot of credits already, I thought testing the player on IRC layout just here, only one time should be sufficient/reasonable, as long as the IRC layout design is not changed radically.

Do you think we should check IRC layout for every scenario besides the other layouts (it should be indeed less confusing and might be fine too, as running Percy will become weekly, not per a commit on `develop`) or explain on the file that `takeSnapshots` does not check the player on IRC layout for the reason above?

Reply:
This makes sense - thanks for the explanation.